pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -881,6 +883,11 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, storagePolicyID string, nodeNam
 		return "", "", err
 	}
 	return deviceName, diskUUID, nil
+	}
+	requestTime := time.Now()
+	diskID, diskUUID, err = attachDiskInternal(vmDiskPath, storagePolicyID, nodeName)
+	recordvSphereMetric(request_attachvolume, requestTime, err)
+	return diskID, diskUUID, err
```

Comment:
Yes, these are operations and not one API call.

I have observed requests for CreateVolumeWithPolicy and CreateVolumeWithRawVSANPolicy are always placed in the bucket with upper bound le="+Inf". Means they take more than 10000 milliseconds. 

> 
> cloudprovider_vsphere_api_request_duration_seconds_bucket{request="CreateVolumeWithPolicy",le="10"} 0
> cloudprovider_vsphere_api_request_duration_seconds_bucket{request="CreateVolumeWithPolicy",le="+Inf"} 1

> cloudprovider_vsphere_api_request_duration_seconds_bucket{request="CreateVolumeWithRawVSANPolicy",le="10"} 0
> cloudprovider_vsphere_api_request_duration_seconds_bucket{request="CreateVolumeWithRawVSANPolicy",le="+Inf"} 1

If eventually everything is going to be placed in `+Inf` upper bound, we cannot use alert to notify performance issue.


We should report metrics for full Operations as well as API call. 
Let me revisit the code, and add more metrics.